blogposts/code-intelligence-with-lsif.md

```diff
@@ -0,0 +1,32 @@
+---
+title: 'Code intelligence with LSIF'
+author: 'Chris Wendt'
+publishDate: 2019-12-01T00:00-07:00
+tags: [
+  "blog"
+]
+slug: code-intelligence-with-lsif
+heroImage: https://about.sourcegraph.com/sourcegraph-mark.png
+published: true
+---
+
+Since the [last code intelligence update](https://about.sourcegraph.com/blog/improving-language-support-in-2019), we shifted our efforts from improving language servers (which are difficult to deploy and scale) to building [basic-code-intel](https://github.com/sourcegraph/sourcegraph-basic-code-intel) (based on ctags and text search) in order to contribute to our early-2019 goal of increasing Sourcegraph adoption. Although basic-code-intel works out of the box, its limited analysis provides at best approximate and fuzzy code intelligence.
```

Comment:
Two suggestions:
1. I think it would be worth elaborating on why we decided to stop focusing on language servers.
2. I am wondering if it would be better to avoid calling it "basic-code-intel" and instead stick to something more descriptive.

For example:

```suggestion
Since the [last code intelligence update](https://about.sourcegraph.com/blog/improving-language-support-in-2019), we have shifted our efforts away from improving language servers. We found that language servers were hard to develop due to the number of build and dependency systems per language, hard to deploy and connect to a Sourcegraph instance, and hard to make fast enough to meet our user's expectations. The Sourcegraph 3.0 release in February included [zero configuration code intelligence for the 20+ most popular languages](https://github.com/sourcegraph/sourcegraph-basic-code-intel) based on ctags and text search. This has provided a much better default experience for Sourcegraph customers, but there is one limitation: results are imprecise unless a customer configures a language server.
```